base/shell.jl

```diff
@@ -88,7 +88,10 @@ function shell_parse(str::AbstractString, interpolate::Bool=true;
                 in_double_quotes = !in_double_quotes
                 i = consume_upto!(arg, s, i, j)
             elseif c == '\\'
```

Comment:
The `\` character has no special meaning inside single quotes
```suggestion
            elseif !in_single_quotes && c == '\\'
```